python/cudf/cudf/core/column/column.py

```diff
@@ -181,6 +179,19 @@ def __init__(
         self.set_base_data(data)
         self.set_base_mask(mask)
 
+    @property
+    def _PANDAS_NA_VALUE(self):
+        """Return appropriate NA value based on dtype."""
+        if cudf.get_option("mode.pandas_compatible"):
+            # In pandas compatibility mode, return pd.NA for all
+            # nullable extension dtypes
+            if is_pandas_nullable_extension_dtype(self.dtype):
+                return pd.NA
```

Comment:
```suggestion
                return self.dtype.na_value
```
May help with pandas 3.0 changes when `pd.StringDtype.na_value` can be `NaN`